ui/app/utils/properties/job-client-status.js

```diff
@@ -26,7 +27,9 @@ export default function jobClientStatus(nodesKey, jobKey) {
 
       // Filter nodes by the datacenters defined in the job.
       const filteredNodes = nodes.filter((n) => {
-        return job.datacenters.indexOf(n.datacenter) >= 0;
+        return job.datacenters.filter((dc) => {
+          return !!matchGlob(dc, n.datacenter);
+        }).length;
```

Comment:
`nit` `non-blocking`:  might want to use `find` in the nested loop for an early exit and remove the need for `.length`.

```
const filteredNodes = nodes.filter((n) => job.datacenters.find((dc) => !!matchGlob(dc, n.datacenter));
```